cmd/storage-rest-client.go

```diff
@@ -227,7 +227,8 @@ func (client *storageRESTClient) NSScanner(ctx context.Context, cache dataUsageC
 		rw.CloseWithError(waitForHTTPStream(respBody, rw))
 	}()
 
-	ms := msgp.NewReader(rr)
+	ms := msgpNewReader(rr)
+	defer readMsgpReaderPool.Put(rr)
```

Comment:
Shouldn't this be 'ms' instead 'rr' ?

Reply:
oh yeah... Hmmm.. going to make this type safe. We don't want this mistake elsewhere